src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py

```diff
@@ -298,8 +304,8 @@ def flexible_server_update_custom_func(cmd, client, instance,
     # High availability can't be updated with existing properties
     high_availability_param = postgresql_flexibleservers.models.HighAvailability()
     if high_availability:
-        if high_availability.lower() == "enabled":
```

Comment:
We use list-skus (locationCapabilities) output to dynamically get region information when running the validator. Please add filter to see whether 'SameZone' HAMode is supported in specific region. You might need to modify get_postgres_list_skus_info in flexibleserver_utils file and add the same logic to MySQL if it's missing there too. If there's any validations that needs to be updated, please do so too.

Reply:
I'm already checking the value of the array `supportedHAMode` in location capabilities response, and I modified the validators logic in `_pg_high_availability_validator`.

Reply:
I confirmed with both feature owners of MySQL and PostgreSQL and SameZone HAMode is alwayds supported.